cloud-sql/mysql/servlet/src/main/java/com/example/cloudsql/ConnectionPoolContextListener.java

```diff
@@ -0,0 +1,142 @@
+    config.setMaximumPoolSize(5);
+    // [END cloud_sql_max_connections]
+
+    // [START cloud_sql_connection_timeout]
+    // setConnectionTimeout is the maximum number of milliseconds to wait for a connection checkout.
+    // Any attempt to retrieve a connection from this pool that exceeds the set limit will throw an
+    // SQLException.
+    config.setConnectionTimeout(10000); // 10 seconds
+    // [END cloud_sql_connection_timeout]
+
+    // [START cloud_sql_connection_backoff]
+    // Hikari automatically delays between failed connection attempts, eventually reaching a
+    // maximum delay of `connectionTimeout / 2` between attempts.
+    // [END cloud_sql_connection_backoff]
+
+    // [START cloud_sql_connection_lifetime]
+    // maxLifetime is the maximum possible lifetime of a connection in the pool. Connections that
+    // live longer than this many milliseconds will be closed and reestablished between uses. This
+    // value should be several minutes shorter than the database's timeout value to avoid unexpected
+    // terminations.
+    config.setMaxLifetime(1800000); // 30 minutes
+    // [END cloud_sql_connection_lifetime]
+
+    // [START cloud_sql_idle_connections]
+    // minimumIdle is the minimum number of idle connections Hikari maintains in the pool.
+    // Additional connections will be established to meet this value unless the pool is full.
+    config.setMinimumIdle(5);
+    // idleTimeout is the maximum amount of time a connection can sit in the pool. Connections that
+    // sit idle for this many milliseconds are retried if minimumIdle is exceeded.
+    config.setIdleTimeout(600000); // 10 minutes
+    // [END cloud_sql_idle_connections]
+    // [END_EXCLUDE]
+
+    // Initialize the connection pool using the configuration object.
+    DataSource pool = new HikariDataSource(config);
+    // [END cloud_sql_mysql_connection_pool]
+    return pool;
+  }
+
+  private void createTableSchema(DataSource pool) {
+    // Safely attempt to create the table schema.
+    try (Connection conn = pool.getConnection()) {
+      PreparedStatement createTableStatement = conn.prepareStatement(
+          "CREATE TABLE IF NOT EXISTS votes ( "
+              + "vote_id SERIAL NOT NULL, time_cast timestamp NOT NULL, canidate CHAR(6) NOT NULL, "
+              + "PRIMARY KEY (vote_id) );"
+      );
+      createTableStatement.execute();
+    } catch (SQLException e) {
+      throw new Error(
+          "Unable to successfully verify table schema. Please double check the steps in the README"
+              + " and restart the application. \n" + e.toString());
+    }
+  }
+
+  @Override
+  public void contextDestroyed(ServletContextEvent event) {
+    // This function is called when the Servlet is destroyed.
+    DataSource pool = (DataSource) event.getServletContext().getAttribute("my-pool");
+    if (pool != null) {
+      try {
+        pool.unwrap(HikariDataSource.class).close();
```

Comment:
Since we know this is already an `HikariDataSource`, it should simply cast to it.

`HikariDataSource pool = (HikariDataSource) ...